### ISSUE TITLE
retentionSize is not taken into effect

### ISSUE THREAD

**Issue by radhupr** [prometheus-operator/prometheus-operator#6854]
### Is there an existing issue for this?

- [X] I have searched the existing issues

## Description

I have below setting in the yaml 
```
prometheus:
 
  agentMode: true
  prometheusSpec:
    # retention: 1h
    retentionSize: 8Gi
```
But the rententionSize is not respected and I have volume filling up issue for prometheus  statefulset.
The total storage size for the volume is set as 10Gi.
## Steps to Reproduce
Using helm chart https://prometheus-community.github.io/helm-charts release : prometheus-operator version: 60.0.0
## Expected Result
retentionsize is respected and added to operator
## Actual Result
The retention size is not respected and volume is filling up.

### Prometheus Operator Version

```shell
chart https://prometheus-community.github.io/helm-charts release : prometheus-operator version: 60.0.0
```

### Kubernetes Version

```yaml
v1.28.3
```

### Kubernetes Cluster Type

kubeadm

### How did you deploy Prometheus-Operator?

helm chart:prometheus-community/kube-prometheus-stack

### prometheus-operator log output

```shell
level=info ts=2024-08-19T18:24:01.69091489Z caller=main.go:199 component=automaxprocs msg="Leaving GOMAXPROCS=32: CPU quota undefined"
level=info ts=2024-08-19T18:24:01.690976562Z caller=main.go:213 msg="Starting Prometheus Operator" version="(version=0.74.0, branch=refs/tags/v0.74.0, revision=cd506c4fb584f40b46163926f01ba24a91f5baf5)"
level=info ts=2024-08-19T18:24:01.691049799Z caller=main.go:214 build_context="(go=go1.22.3, platform=linux/amd64, user=Action-Run-ID-9130603252, date=20240517-15:27:00, tags=unknown)"
level=info ts=2024-08-19T18:24:01.69107945Z caller=main.go:215 feature_gates=
level=info ts=2024-08-19T18:24:01.691127075Z caller=main.go:226 msg="namespaces filtering configuration " config="{allow_list=\"\",deny_list=\"\",prometheus_allow_list=\"\",alertmanager_allow_list=\"\",alertmanagerconfig_allow_list=\"\",thanosruler_allow_list=\"\"}"
level=info ts=2024-08-19T18:24:01.781741581Z caller=main.go:255 msg="connection established" cluster-version=v1.28.3
level=info ts=2024-08-19T18:24:02.260667045Z caller=operator.go:335 component=prometheus-controller msg="Kubernetes API capabilities" endpointslices=true
level=info ts=2024-08-19T18:24:02.500561418Z caller=operator.go:320 component=prometheusagent-controller msg="Kubernetes API capabilities" endpointslices=true
level=warn ts=2024-08-19T18:24:02.933905161Z caller=server.go:160 msg="server TLS client verification disabled" client_ca_file=/etc/tls/private/tls-ca.crt err="stat /etc/tls/private/tls-ca.crt: no such file or directory"
level=info ts=2024-08-19T18:24:02.935314938Z caller=server.go:300 msg="starting secure server" address=[::]:10250 http2=false
level=info ts=2024-08-19T18:24:03.135811021Z caller=operator.go:283 component=thanos-controller msg="successfully synced all caches"
level=info ts=2024-08-19T18:24:03.236384067Z caller=operator.go:429 component=prometheusagent-controller msg="successfully synced all caches"
level=info ts=2024-08-19T18:24:03.236591404Z caller=operator.go:313 component=alertmanager-controller msg="successfully synced all caches"
level=info ts=2024-08-19T18:24:03.236599623Z caller=operator.go:392 component=prometheus-controller msg="successfully synced all caches"
level=info ts=2024-08-19T18:24:03.236623333Z caller=operator.go:563 component=prometheusagent-controller key=prometheus-operator/prometheus-prometheus msg="sync prometheus"
level=info ts=2024-08-19T18:24:03.664000605Z caller=operator.go:563 component=prometheusagent-controller key=prometheus-operator/prometheus-prometheus msg="sync prometheus"
```

**Comment by chengjoey** [comment]
`retentionSize` acts on prometheus server args, describe sts and view the parameters of args `storage.tsdb.retention.size`

**Comment by radhupr** [comment]
Here is the output of statefulset
```
Name:               prom-agent-prometheus-prometheus
Namespace:          prometheus-operator
CreationTimestamp:  Tue, 13 Aug 2024 17:50:58 +0300
Selector:           app.kubernetes.io/instance=prometheus-prometheus,app.kubernetes.io/managed-by=prometheus-operator,app.kubernetes.io/name=prometheus-agent,operator.prometheus.io/name=prometheus-prometheus,operator.prometheus.io/shard=0
Labels:             app=prometheus-prometheus
                    app.kubernetes.io/instance=prometheus-operator
                    app.kubernetes.io/managed-by=Helm
                    app.kubernetes.io/part-of=prometheus
                    app.kubernetes.io/version=60.0.0
                    argocd.argoproj.io/instance=prometheus-operator-staging-eu
                    chart=kube-prometheus-stack-60.0.0
                    heritage=Helm
                    managed-by=prometheus-operator
                    operator.prometheus.io/mode=agent
                    operator.prometheus.io/name=prometheus-prometheus
                    operator.prometheus.io/shard=0
                    release=prometheus-operator
Annotations:        prometheus-operator-input-hash: 5786596280432291425
Replicas:           2 desired | 2 total
Update Strategy:    RollingUpdate
Pods Status:        2 Running / 0 Waiting / 0 Succeeded / 0 Failed
Pod Template:
  Labels:           app.kubernetes.io/instance=prometheus-prometheus
                    app.kubernetes.io/managed-by=prometheus-operator
                    app.kubernetes.io/name=prometheus-agent
                    app.kubernetes.io/version=2.52.0
                    operator.prometheus.io/name=prometheus-prometheus
                    operator.prometheus.io/shard=0
  Annotations:      kubectl.kubernetes.io/default-container: prometheus
                    kubectl.kubernetes.io/restartedAt: 2024-08-19T08:58:54Z
                    linkerd.io/inject: enabled
  Service Account:  prometheus-prometheus
  Init Containers:
   init-config-reloader:
    Image:      quay.io/prometheus-operator/prometheus-config-reloader:v0.74.0
    Port:       8080/TCP
    Host Port:  0/TCP
    Command:
      /bin/prometheus-config-reloader
    Args:
      --watch-interval=0
      --listen-address=:8080
      --config-file=/etc/prometheus/config/prometheus.yaml.gz
      --config-envsubst-file=/etc/prometheus/config_out/prometheus.env.yaml
    Limits:
      cpu:     30m
      memory:  300Mi
    Requests:
      cpu:     10m
      memory:  150Mi
    Environment:
      POD_NAME:   (v1:metadata.name)
      SHARD:     0
    Mounts:
      /etc/prometheus/config from config (rw)
      /etc/prometheus/config_out from config-out (rw)
  Containers:
   prometheus:
    Image:      quay.io/prometheus/prometheus:v2.52.0
    Port:       9090/TCP
    Host Port:  0/TCP
    Args:
      --web.console.templates=/etc/prometheus/consoles
      --web.console.libraries=/etc/prometheus/console_libraries
      --config.file=/etc/prometheus/config_out/prometheus.env.yaml
      --web.enable-lifecycle
      --enable-feature=agent
      --web.external-url=http://prometheus-prometheus.prometheus-operator:9090
      --web.route-prefix=/
      --storage.agent.path=/prometheus
      --storage.agent.wal-compression
      --web.config.file=/etc/prometheus/web_config/web-config.yaml
    Limits:
      cpu:     1
      memory:  15Gi
    Requests:
      cpu:        500m
      memory:     10Gi
    Liveness:     http-get http://:http-web/-/healthy delay=0s timeout=3s period=5s #success=1 #failure=6
    Readiness:    http-get http://:http-web/-/ready delay=0s timeout=3s period=5s #success=1 #failure=3
    Startup:      http-get http://:http-web/-/ready delay=0s timeout=3s period=15s #success=1 #failure=60
    Environment:  <none>
    Mounts:
      /etc/prometheus/certs from tls-assets (ro)
      /etc/prometheus/config_out from config-out (ro)
      /etc/prometheus/secrets/prometheus-sp-oauth from secret-prometheus-sp-oauth (ro)
      /etc/prometheus/web_config/web-config.yaml from web-config (ro,path="web-config.yaml")
      /prometheus from prom-agent-prometheus-prometheus-db (rw,path="prometheus-db")
   config-reloader:
    Image:      quay.io/prometheus-operator/prometheus-config-reloader:v0.74.0
    Port:       8080/TCP
    Host Port:  0/TCP
    Command:
      /bin/prometheus-config-reloader
    Args:
      --listen-address=:8080
      --reload-url=http://127.0.0.1:9090/-/reload
      --config-file=/etc/prometheus/config/prometheus.yaml.gz
      --config-envsubst-file=/etc/prometheus/config_out/prometheus.env.yaml
    Limits:
      cpu:     30m
      memory:  300Mi
    Requests:
      cpu:     10m
      memory:  150Mi
    Environment:
      POD_NAME:   (v1:metadata.name)
      SHARD:     0
    Mounts:
      /etc/prometheus/config from config (rw)
      /etc/prometheus/config_out from config-out (rw)
  Volumes:
   config:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  prom-agent-prometheus-prometheus
    Optional:    false
   tls-assets:
    Type:                Projected (a volume that contains injected data from multiple sources)
    SecretName:          prom-agent-prometheus-prometheus-tls-assets-0
    SecretOptionalName:  <nil>
   config-out:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:     Memory
    SizeLimit:  <unset>
   secret-prometheus-sp-oauth:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  prometheus-sp-oauth
    Optional:    false
   web-config:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  prom-agent-prometheus-prometheus-web-config
    Optional:    false
Volume Claims:
  Name:          prom-agent-prometheus-prometheus-db
  StorageClass:  longhorn
  Labels:        <none>
  Annotations:   <none>
  Capacity:      10Gi
  Access Modes:  [ReadWriteOnce]
```

**Comment by simonpasquier** [comment]
PrometheusAgent doesn't support the retention settings because it doesn't use a fully-fledged TSDB.

**Comment by simonpasquier** [comment]
TL;DR: it's more a question/issue for the Helm charts community.

**Comment by radhupr** [comment]
> PrometheusAgent doesn't support the retention settings because it doesn't use a fully-fledged TSDB.

If it doesnt support then why would helm chart give an option to configure those parameters 🤔

**Comment by slashpai** [comment]
> > PrometheusAgent doesn't support the retention settings because it doesn't use a fully-fledged TSDB.
> 
> If it doesnt support then why would helm chart give an option to configure those parameters 🤔

You need to ask in https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack
